### ISSUE TITLE
Routing issue with language and currencies

### ISSUE THREAD

**Issue by patrickfabrizius** [coreshop/CoreShop#82]
Steps to reproduce: Clean install, create a product, change currency to SEK, go to front end, everything is now shown in SEK. But if I select SEK from the currency nav, I am redired to http://root/productName~c11 . 

Related note: If I do the same from the cart, I am instead redired to http://myhost/root/cart/%action/?currency=71 .

Related note 2: When I click on the cart, the site language URL prefix is changed from the default /en to /de .

**Comment by dpfaffenbauer** [comment]
I did find an issue with changing the currency within the cart or checkout process.

this is fixed wthin https://github.com/coreshop/website-example/commit/37ac5a9369ecffd1c862869f44a3eb14da96ad63 and 7b9b1cd

I also did find the bug why it changes your language. (https://github.com/coreshop/website-example/commit/c2c899c27e4efe981bf14467a07810a625c0cc64)

But i could not find the problem with http://root/productName~c11. Where does that root come from?

**Comment by patrickfabrizius** [comment]
I can reproduce it by clicking "Shop" (home) and then selecting currency (or click on a product first and then currency, if you meant the productName specifically). See attached screen dump:

![image](https://cloud.githubusercontent.com/assets/6759985/13318404/d34c6bba-dbba-11e5-9f5e-61967f663256.png)

**Comment by patrickfabrizius** [comment]
(Only difference with this one related to the one you already fixed(?) seems to be a missing slash for the URI?)

**Comment by dpfaffenbauer** [comment]
this is a weird behaviour. i think the problem is that the /en document has no language set. 

Since i never really like the behaviour for chaning the language I did some changes.

i added a HelperController to change the currency. could you re-pull CoreShop and the Example Website and re-try it?

**Comment by patrickfabrizius** [comment]
I updated CoreShop plugin and website/ but now any frontend page throws an exception: 

`Route 'coreshop_change_currency' for building the URL not found#0 [internal function]: Pimcore\View\Helper\Url->url(Array, 'coreshop_change...') #1 /Users/patrick/Sites/internt/pimcore/vendor/zendframework/zendframework1/library/Zend/View/Abstract.php(350): call_user_func_array(Array, Array) #2 /Users/patrick/Sites/internt/pimcore/pimcore/lib/Pimcore/View.php(373): Zend_View_Abstract->__call('url', Array) #3 /Users/patrick/Sites/internt/pimcore/website/views/scripts/coreshop/template/default/layouts/shop.php(148): Pimcore\View->__call('url', Array) #4 /Users/patrick/Sites/internt/pimcore/website/views/scripts/coreshop/template/default/layouts/shop.php(148): Pimcore\View->url(Array, 'coreshop_change...') #5 /Users/patrick/Sites/internt/pimcore/vendor/zendframework/zendframework1/library/Zend/View.php(157): include('/Users/patrick/...') #6 /Users/patrick/Sites/internt/pimcore/vendor/zendframework/zendframework1/library/Zend/View/Abstract.php(888): Zend_View->_run('/Users/patrick/...') #7 /Users/patrick/Sites/internt/pimcore/vendor/zendframework/zendframework1/library/Zend/Layout.php(796): Zend_View_Abstract->render('shop.php') #8 /Users/patrick/Sites/internt/pimcore/vendor/zendframework/zendframework1/library/Zend/Layout/Controller/Plugin/Layout.php(143): Zend_Layout->render() #9 /Users/patrick/Sites/internt/pimcore/vendor/zendframework/zendframework1/library/Zend/Controller/Plugin/Broker.php(333): Zend_Layout_Controller_Plugin_Layout->postDispatch(Object(Zend_Controller_Request_Http)) #10 /Users/patrick/Sites/internt/pimcore/vendor/zendframework/zendframework1/library/Zend/Controller/Front.php(965): Zend_Controller_Plugin_Broker->postDispatch(Object(Zend_Controller_Request_Http)) #11 /Users/patrick/Sites/internt/pimcore/pimcore/lib/Pimcore.php(269): Zend_Controller_Front->dispatch() #12 /Users/patrick/Sites/internt/pimcore/index.php(16): Pimcore::run() #13 {main}
`

I tried wiping everything, did a clean Pimcore nightly install, and copied CoreShop + website again. In extension manager the install completely breaks Pimcore (there is an exception shown but unfortunately I didn't have the time to copy it until a Pimcore ping makes Pimcore go bazonkas). Both admin/frontend now just shows a 500 (no stack trace). I found this in debug.log: 

`[2016-02-25 16:43:26] core.WARNING: exception 'Exception' with message 'Configuration with key: SYSTEM.ISINSTALLED does not exist' in /Users/patrick/Sites/internt/pimcore/plugins/CoreShop/lib/CoreShop/Model/Configuration/Dao.php:73 Stack trace: #0 /Users/patrick/Sites/internt/pimcore/plugins/CoreShop/lib/CoreShop/Model/Configuration.php(106): CoreShop\Model\Configuration\Dao->getByKey('SYSTEM.ISINSTAL...') #1 /Users/patrick/Sites/internt/pimcore/plugins/CoreShop/lib/CoreShop/Model/Configuration.php(134): CoreShop\Model\Configuration::get('SYSTEM.ISINSTAL...', true) #2 /Users/patrick/Sites/internt/pimcore/plugins/CoreShop/lib/CoreShop/Plugin/Install.php(641): CoreShop\Model\Configuration::set('SYSTEM.ISINSTAL...', false) #3 /Users/patrick/Sites/internt/pimcore/plugins/CoreShop/lib/CoreShop/Plugin.php(189): CoreShop\Plugin\Install->createConfig() #4 /Users/patrick/Sites/internt/pimcore/pimcore/modules/extensionmanager/controllers/AdminController.php(122): CoreShop\Plugin::install() #5 /Users/patrick/Sites/internt/pimcore/vendor/zendframework/zendframework1/library/Zend/Controller/Action.php(516): Extensionmanager_AdminController->installAction() #6 /Users/patrick/Sites/internt/pimcore/vendor/zendframework/zendframework1/library/Zend/Controller/Dispatcher/Standard.php(308): Zend_Controller_Action->dispatch('installAction') #7 /Users/patrick/Sites/internt/pimcore/vendor/zendframework/zendframework1/library/Zend/Controller/Front.php(954): Zend_Controller_Dispatcher_Standard->dispatch(Object(Zend_Controller_Request_Http), Object(Zend_Controller_Response_Http)) #8 /Users/patrick/Sites/internt/pimcore/pimcore/lib/Pimcore.php(269): Zend_Controller_Front->dispatch() #9 /Users/patrick/Sites/internt/pimcore/index.php(16): Pimcore::run() #10 {main} {"origin":"CoreShop\\Model\\Configuration::get() on line 108","memory":" 18 MB"} []
`

**Comment by dpfaffenbauer** [comment]
First Problem:
You did pull the files via git didnt you? I created a new static route for changing the currency. See (CoreShop/lib/install/staticroutes.xml)

Second Problem:
This happens when CoreShop cant find the Template, make source you used the latest website-example where the coreshop files are located in website/views/scripts/coreshop

**Comment by patrickfabrizius** [comment]
Yes, I pulled it directly from git to the plugins directory, that didn't work. On the second attempt though I think I copied in an old version (since I wiped everything). I just tried again with fresh pulls and now it works :) Not able to reproduce the problem with the currency selector. 

Trying to complete a checkout though, I get caught in a redirect loop when trying to register as customer.
